### PR TITLE
Baroclinic example

### PR DESCRIPTION
**Pull request by Witt-D**
Added baroclinic wave script

### CODE REVIEW

File: compressible/baroclinic_wave.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
+"""
+The deep atmosphere dry baroclinic wave Test case from Ullich et al. 2013
+
+Variable height: Applies a non uniform height field.
+                 Default = True
+Alpha:           Adjusts the ratio of implicit to explicit in the solver.
+                 Default = 0.5
```
Comment (JDBetteridge) [inline compressible/baroclinic_wave.py:7]: Remove these as they are documented in the arguments

File: compressible/baroclinic_wave.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    # assign reference profiles
+    stepper.set_reference_profiles([('rho', rho_b),
+                                    ('theta', theta_b)])
+
+    stepper.run(t=0, tmax=tmax)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(
+        description="The deep atmosphere dry baroclinic wave Test case from Ullich et al. 2013",
```
Comment (JDBetteridge) [inline compressible/baroclinic_wave.py:211]:
```suggestion
        description=__doc__,
```
This is what I was trying to remember when I was live coding things!

File: compressible/test_compressible.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+from dry_baroclinic_sphere import dry_baroclinic_sphere
+
+def test_dry_baroclinic_sphere():
+    dry_baroclinic_sphere(4, 3, 900.0, 1800.0, 2)
```
Comment (jshipton) [inline compressible/test_compressible.py:4]: adding the parameters here would be nice

**Review by jshipton — APPROVED** [review]
looks great!

File: compressible/makefile [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+dry_baroclinic_sphere:
+	mpiexec -n 16 python3 dry_baroclinic_sphere.py
+
+baroclinic_image:
```
Comment (JDBetteridge) [inline compressible/makefile:5]: Remove or populate this recipe

File: compressible/readme.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Compressible cases studies
+
+## Dry Baroclinic Wave on the Sphere
+
```
Comment (JDBetteridge) [inline compressible/readme.md:4]: This is just a repetition of the docstring, do you really want to do this?

File: compressible/test_compressible.py (hunk 3 above)
Comment (JDBetteridge) [inline compressible/test_compressible.py:4]: Adding the argument names makes this a bit clearer ie:
```python
dry_baroclinic_sphere(
    ncell_per_edge=4,
    nlayers=3,
    dt=900,
    tmax=1800,
    dumpfreq=2
)
```

File: compressible/dry_baroclinic_sphere.py [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+                       exp, Constant, Function, as_vector, acos, errornorm,
+                       norm, le, ge, conditional, NonlinearVariationalProblem,
+                       NonlinearVariationalSolver, TestFunction, inner, dx)
+from gusto import (Domain, GeneralCubedSphereMesh, CompressibleParameters,
+                   CompressibleEulerEquations, OutputParameters, IO,
+                   EmbeddedDGOptions, SSPRK3, DGUpwind, CompressibleSolver,
+                   SemiImplicitQuasiNewton, lonlatr_from_xyz,
+                   xyz_vector_from_lonlatr, compressible_hydrostatic_balance)
+
+def dry_baroclinic_sphere(ncell_per_edge, nlayers, dt, tmax, dumpfreq, dirname):
```
Comment (JDBetteridge) [inline compressible/dry_baroclinic_sphere.py:23]: Do you want to specify defaults here, for the case where this function is imported elsewhere? I'm not sure it's 100% necessary, it depends on the intended use.

File: compressible/readme.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+The deep atmosphere dry baroclinic wave from Ullrich et al. 2014:
+''A proposed baroclinic wave test case for deep- and shallow-atmosphere
+dynamical cores'', QJRMS.
+
+This is a 3D test on the sphere, with an initial state that is in unsteady
+balance, with a perturbation added to the wind.
+
+This setup uses a cubed-sphere with the order 1 finite elements.
```
Comment (JDBetteridge) [inline compressible/readme.md:12]: I could write a simple template which could autogenerate this + the help output, if that is desirable

**Review by JDBetteridge — APPROVED** [review]

**Review by tommbendall — APPROVED** [review]
This looks really neat now!

**Pull request merged**